lib/zip/extra_field/universal_time.rb

```diff
@@ -8,20 +8,52 @@ def initialize(binstr = nil)
       @ctime = nil
       @mtime = nil
       @atime = nil
-      @flag  = nil
-      binstr && merge(binstr)
+      @flag  = 0
+
+      merge(binstr) unless binstr.nil?
+    end
+
+    attr_reader :atime, :ctime, :mtime, :flag
+
+    def atime=(time)
+      time.nil? ? @flag &= 0b101 : @flag |= 0b010
```

Comment:
Maybe it's worth extracting these masks into constants? Something like
```rb
ATIME_MASK = 0b010
@flag = time.nil? ? @flag & ~ATIME_MASK : @flag | ATIME_MASK
```

Reply:
Agreed. I'll do that.